File: playwright-python-tests/page-objects/LoginPage.ts
import { Page, Locator } from '@playwright/test';

export class LoginPage {
  readonly page: Page;
  readonly usernameField: Locator;
  readonly passwordField: Locator;
  readonly submitBtn: Locator;
  readonly errorMessage: Locator;
  readonly forgotLink: Locator;

  constructor(page: Page) {
    this.page = page;
    this.usernameField = page.locator('#login_field');
    this.passwordField = page.locator('#password');
    this.submitBtn = page.locator("input[type='submit']");
    this.errorMessage = page.locator('.flash-error');
    this.forgotLink = page.locator("a:has-text('Forgot password?')");
  }

  async navigate(): Promise<void> {
    await this.page.goto('https://github.com/login', { waitUntil: 'networkidle' });
  }

  async login(user: string, pass: string): Promise<void> {
    await this.navigate();
    await this.usernameField.fill(user);
    await this.passwordField.fill(pass);
    await this.submitBtn.click();
  }
}

File: playwright-python-tests/page-objects/login_page.py
from .base_page import BasePage

class LoginPage(BasePage):
    """LoginPage maps the login interface targets."""
    def __init__(self, page):
        super().__init__(page)
        self.url = "https://github.com/login"
        self.username_field = "#login_field"
        self.password_field = "#password"
        self.submit_btn = "input[type='submit']"
        self.error_message = ".flash-error"
        self.forgot_link = "a:has-text('Forgot password?')"

    def login(self, user: str, password: str) -> None:
        self.navigate(self.url)
        self.fill_input(self.username_field, user)
        self.fill_input(self.password_field, password)
        self.click(self.submit_btn)
